Code cleanup in the tests

lg2de opened the pull request:
Update of test code according to the design guide, fixes #325 .

In `AwesomeAssertions.Equivalency.Specs` I applied the "because" and the "AAA comment" pattern. This causes a lot of changes need to be reviewed.
So, `AwesomeAssertions.Specs` I applied only the "because" pattern.

## Legal checklist

* [x] This work is entirely original, it is not derived from any existing code incompatible with the Apache 2.0 License, like FluentAssertions.

lg2de committed:
Update "because" and "AAA comment" pattern in Equivalency.Specs

lg2de committed:
Disable rule MA0007

lg2de committed:
Use compact format for test code

lg2de commented:
Dear @AwesomeAssertions/collaborators , please have a look on my WIP. Do you agree with my attempt to unify and compact the test code?

cbersch commented on Tests/.editorconfig:
```diff
@@ -186,6 +188,9 @@ dotnet_diagnostic.RCS1241.severity = none
 # Use Array.Empty<T>(). We don't care.
 dotnet_diagnostic.MA0005.severity = none
 
+# Add comma to multi-line initializer. We don't care.
```
For me, that's a coding style decision which should only be made in the toplevel `.editorconfig`.

lg2de replied:
Ok, I'll move it to the top-level file.

lg2de committed:
Update design guide regarding AAA comments

lg2de committed:
Update "because" pattern in Collections Specs

lg2de committed:
Update Events Specs

lg2de commented on Tests/AwesomeAssertions.Specs/Collections/GenericCollectionAssertionOfStringSpecs.Equal.cs:
```diff
@@ -98,9 +98,9 @@ public void When_asserting_collections_to_be_equal_but_expected_collection_is_nu
-                collection.Should().Equal(collection1, "because we want to test the behaviour with a null subject");
+            Action act = () => collection.Should().Equal(collection1, "we want to test the {0} message", "failure");
 
+            // TODO should the message contain the because text?
```
Show the because argument is in case of "Cannot compare collection"?

lg2de reviewed (commented):
Dear @AwesomeAssertions/collaborators , while continue working on the because arguments (only!), I found some more inconsistencies: The because argument is specified but not used in a special form of error message.
I'm unsure whether these use cases should use the because argument.
Please, help me to decide!

lg2de committed:
Update Exceptions Specs

lg2de committed:
Update Execution Specs

lg2de committed:
Update Numeric Specs

lg2de committed:
Fix S4144 "Methods should not have identical implementations"

https://sonarcloud.io/organizations/awesomeassertions/rules?open=csharpsquid%3AS4144&q=S4144

lg2de committed:
Add tests to ensure that "because" is always added

lg2de committed:
Update Primitives Specs

ScarletKuro commented on Tests/AwesomeAssertions.Specs/Collections/GenericCollectionAssertionOfStringSpecs.Contain.cs:
```diff
@@ -46,7 +46,25 @@ public void
 
             // Act / Assert
-            actual.Should().Contain(expected, "they are in the collection");
+            // TODO The be because arguments cannot be checked in successful test. But, it seems to affect the coverage.
```
I tried to get report files with because and without on this 49 line and compare them, and the files are identical.
Seems it's just a flaky baseline?

lg2de replied:
> Is it intended that you have two similar links?

No. Thanks! I've fixed that.

lg2de committed:
Update Specialized Specs

lg2de committed:
Update Streams Specs

lg2de committed:
Update Types Specs

lg2de committed:
Update Xml Specs

lg2de committed:
Update ExtensibilitySpecs

cbersch commented on Tests/AwesomeAssertions.Equivalency.Specs/NestedPropertiesSpecs.cs:
```diff
@@ -93,58 +77,47 @@ public class Item
-                - Without automatic conversion.
-
-                """);
+                         Expected property subject.Level.Text to be the same string, but they differ at index 5:
```
Here, I would keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Equivalency.Specs/SelectionRulesSpecs.Basic.cs:
```diff
@@ -93,206 +62,127 @@ public class ClassWithIndexer
-                    *property subject.Property2*to be *"B" *"2"
-                    *property subject.SubType1.SubProperty1*to be *"C" * "3"*
-                    """);
+                             *property subject.Property1*to be *"A" *"1"
```
Here, I would keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Numeric/NullableNumericAssertionSpecs.BeGreaterThan.cs:
```diff
@@ -137,7 +137,7 @@ public void To_test_the_null_path_for_difference_on_nullable_decimal()
         {
             // Arrange
-            var value = (short?)11;
+            var value = 11;
```
You must keep the cast to `(short)`. Yes, the `?` was wrong.

lg2de replied:
Yes!

cbersch commented on Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Be.cs:
```diff
@@ -80,12 +81,12 @@ public void When_the_expected_string_is_shorter_than_the_actual_string_it_should
-                  "AB"
-                     ↑ (expected).
-                """);
+                                                                           Expected string to be the same string, but they differ at index 2:
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Be.cs:
```diff
@@ -96,12 +97,12 @@ public void When_the_expected_string_is_longer_than_the_actual_string_it_should_
-                  "ABC"
-                     ↑ (expected).
-                """);
+                                                                           Expected string to be the same string, but they differ at index 2:
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Be.cs:
```diff
@@ -203,12 +204,12 @@ public void When_two_strings_differ_and_one_of_them_is_long_it_should_display_bo
-                  "0987654321"
-                   ↑ (expected).
-                """);
+                                                             Expected string to be the same string, but they differ at index 0:
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Be.cs:
```diff
@@ -219,16 +220,17 @@ public void When_two_strings_differ_and_one_of_them_is_multiline_it_should_displ
-                  "A\r\nC"
-                        ↑ (expected).
-                """);
+                                                                           Expected string to be the same string, but they differ on line 2 and column 1 (index 3):
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Be.cs:
```diff
@@ -238,12 +240,12 @@ public void When_two_strings_differ_and_contain_white_spaces_to_escape_these_are
-                  "…and \t\r\nDIFF\r\n\t"
-                              ↑ (expected).
-                """);
+                                                                           Expected subject to be the same string, but they differ on line 3 and column 1 (index 66):
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Primitives/StringAssertionSpecs.Be.cs:
```diff
@@ -368,36 +380,37 @@ public void Mismatches_in_multiline_text_includes_the_line_number()
-            @startuml
-            Alice -> Bob : Authentication Request
-            Bob --> Alice : Authentication Response
+                          @startuml
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch commented on Tests/AwesomeAssertions.Specs/Types/MethodInfoSelectorAssertionSpecs.cs:
```diff
@@ -386,16 +384,16 @@ public void When_asserting_methods_are_not_async_but_async_methods_are_found_it_
-                    Task AwesomeAssertions.Specs.Types.ClassWithAllMethodsAsync.InternalAsyncDoNothing
-                    Task AwesomeAssertions.Specs.Types.ClassWithAllMethodsAsync.ProtectedAsyncDoNothing
-                    """);
+                             Expected all selected methods not to be async because we want to test the failure message, but the following methods are:
```
Keep the original indentation (level + 1 with respect to the previous line.)

cbersch reviewed (changes requested):
Nice big piece of work!
9 out of 10 comments are regarding indentation of raw string literals.
Being able to keep the indentation low is the main advantage of raw string literals in my opinion:
```csharp
.WithMessage("""
    I start with
    minor indentation
    """);
```
If we prefer having the initial `"""` on the next line or not isn't important to me.

lg2de committed:
Configure indention of raw string literals
